### PR TITLE
build(test-dds-utils): use fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to fluid-tsc

### CODE REVIEW

File: packages/dds/test-dds-utils/package.json [hunk 1]
```diff
@@ -33,11 +67,13 @@
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../.prettierignore",
 		"test": "npm run test:mocha",
 		"test:coverage": "c8 npm test",
-		"test:mocha": "npm run test:mocha:base -- \"dist/test/**/*.spec.*js\"",
+		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
```
Comment (sonalideshpandemsft) [inline packages/dds/test-dds-utils/package.json:70]: nit: skip cjs testing

**Review by sonalideshpandemsft — APPROVED** [review]

File: packages/dds/test-dds-utils/package.json (hunk 1 above)
Comment (jason-ha) [reply packages/dds/test-dds-utils/package.json:70]: As a test util used in other test, probably should test both.

File: packages/dds/test-dds-utils/package.json [hunk 2]
```diff
@@ -92,22 +128,18 @@
...
-					"api-extractor:commonjs"
+					"api-extractor:commonjs",
+					"api-extractor:esnext"
 				],
 				"script": false
-			},
-			"tsc": [
-				"...",
-				"@fluid-private/stochastic-test-utils#build:test"
-			]
```
Comment (jason-ha) [inline packages/dds/test-dds-utils/package.json:110]: This should not have been removed. Yes, it is odd, but stochastic-test-utils builds test utils as part of build:test and test-dds-utils consumes those as part of product build.
We ought to edit fluid build tools to understand exports and which tasks produce them, etc.

File: packages/dds/test-dds-utils/package.json [hunk 3]
```diff
@@ -33,11 +67,13 @@
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../.prettierignore",
 		"test": "npm run test:mocha",
 		"test:coverage": "c8 npm test",
-		"test:mocha": "npm run test:mocha:base -- \"dist/test/**/*.spec.*js\"",
+		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
 		"test:mocha:base": "mocha --ignore \"dist/test/types/*\" -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:mocha:cjs": "mocha --recursive \"dist/test/**/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
```
Comment (jason-ha) [inline packages/dds/test-dds-utils/package.json:72]: test:mocha was using test:mocha:base

**Review by jason-ha — COMMENTED** [review]
Proposed fixups (+ a little more) in #19932

**Pull request merged**